compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaLexer.java

```diff
@@ -1075,8 +1075,57 @@ private boolean isIdentifierInitialChar(int c) {
             return true;
         }
 
-        // TODO: if (UnicodeIdentifierChar) return false;
-        return false;
+        return isUnicodeIdentifierChar(c);
+    }
+
+    /**
+     * <p>
+     * Check whether a given char is a unicode identifier char.
+     * </p>
+     * <code> UnicodeIdentifierChar := ^ ( AsciiChar | UnicodeNonIdentifierChar ) </code>
+     *
+     * @param c character to check
+     * @return <code>true</code>, if the character is a unicode identifier char. <code>false</code> otherwise.
+     */
+    private boolean isUnicodeIdentifierChar(int c) {
+        //check Ascii char range
+        if (0x0000 <= c && c <= 0x007F) {
+            return false;
+        }
+
+        //check unicode private use char
+        if (isUnicodePrivateUseChar(c) || isUnicodePatternWhiteSpaceChar(c)) {
+            return false;
+        }
+
+        //TODO: if (UnicodePatternSyntaxChar) return false
+        return true;
```

Comment:
Can we simplify the above two return statements? 
i.e. `return !isUnicodePrivateUseChar(c) && !isUnicodePatternWhiteSpaceChar(c);`

Reply:
I thought these checks are a bit more clear to compare with the specs in this way. And I referred the implementations of similar logics. [[1]](https://github.com/ballerina-platform/ballerina-lang/blob/cf98e6cf208a5f211e33f0a04f42d22d6118580a/compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaLexer.java#L1064)